packages/core/src/types.ts

```diff
@@ -1761,7 +1761,7 @@ export type ActorRefFrom<T> = ReturnTypeOrValue<T> extends infer R
         >
       >
     : R extends Promise<infer U>
-    ? ActorRef<never, U>
```

Comment:
Using `never` is disastrous for the assignability of our types and I have no idea how to make it work, see the playground [here](https://www.typescriptlang.org/play?#code/C4TwDgpgBAQhAWBDAbgSwPYCcA8AVAoshAHbBQQAewJAJgM5QDeUokAXFHcJqsQOZQAvgBoouAHxMAUFFktMiYnVTAMxDgAoIyDgSKkAlFAC8k3FMFSpNCAGMANokzRb6JWTCZ0AW1R0IHHBIaFjYxNoQmKLEAK7eAEaR4lYA9ClQAALAdAC0lJC2wHmYXphSnj5+EAB03IrKqm4azKwBUADk1FztQgZWNg5O0ABmMcSFapxgiADuxHhBKBiY5FS0DIshOIogojvi4hrxupvLBhzI6Kg0UmlQ4UQrfvfoZIh0ynzEiPH20MDoKA7YDwXgCRLKGwMB6RUQRYgvN4fVBfH5-FiAnZSOjTOYaCq+fx9ax2RzOKCjcaNBE42bEABMCwQSywq2oxHosGZWzCESiQOIIAORxO3LOFyuN1pc3p+K8hIgxIGZJcbi4UG8iFsoPCgTFoRa4Da7QAkgA5ADC7VEyEQ9hibViCUiQlEzFcpEowA47vQY299ziiRWgiEyWlDI0mu1vEVqXSOUTVlaUAAgoUsAAlCDDPCEEhkL3rJgsI0cLg8fhhkzSOScWiabS6fOGEySS7XCzJo1c4LLADMef0hbWHIYhvYnG4YNdYkkjBkcjqShUakbOjELeAon89mGHHTAMw2dzegL4iMpjEXZJg3JHvV0Z1EH7er7WEHE+N5qtNrtDo4J1g1nX1PSoH0oFcf1AKDF1Q0EZJbxVCkxgmNwpjpQdcFOVkizHXsWUwQcYX5fZDmOMQcMwc4oA7KVcWIfsoy1Z9+wMIA).

We can revisitr this later, although a promise-based behavior should just ignore whatever it receives, so it conceptually can receive just any event (it is capable of ignoring everything that it receives). So it isn't unsound/incorrect - it's just that perhaps somebody might miss a problem in their code because this is permissive.

Reply:
Using separate generics solves this particular problem - [TS playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggxsA9gJwEoQGYB4AqBRANwgDtgoIAPYEgEwGcoBvKUSALijuGQEtiBzKAF8AfFAC8TAFBRZnWhwAUEAh3xFSASgliCiHjSlCpU1tABCEABYBDAjxQBmXIRJlK1YvSYtwEDly8AsIANFA4YpKMMnLcNsR0PMAOxEoqaq6kYXQQADYYHPBIaJguGsAi2uJiOEYmNBBwuTbI0HCICWQAtjZwVnwQjhyWtvZOWMxmHADkAJIAcgDC02EENrkArv5QxBtdAEYQyKE+7aQeHMztG6QcuwdHwsIi9Y3NrVAYNwgpnGA2AHdiM51G5yFRaAxJn4Atw+IIhGEIop9sNrHYHMgQZlgEjKhw9AYpHR-kDHIoen0Bo5NK8mi02h0uFAwMhEF0eDkhlARhjxsQVEcwvdDsgXiTAcDFKz2ZzBpogA)